### ISSUE TITLE
RethinkDNS + TorBot + Nextdns(private dns)

### ISSUE THREAD

**Issue by evilprime** [celzero/rethink-app#576]
Hi guys, first of all, great app much appreciated! 

My problem though, i am running the Rethinkdns app on my Android 11 phone, in combination with the Torbot, and private dns settings in my network options pointing to a nextdns subscription. 

Inet is not working. When opening my Rethinkdns app it says "protected with TOR & private dns" ..all looks good, but there is no inet. Not sure how to troubleshoot this. I've configured Rethink to only use dns and this disabled private dns, and then it works. The only think actually what doesn't work is this specific combination of Rethinkdns + tor + nextdns private dns. 

is there anything i should take into account or to check or some settings i am not seeing? or is this just by design? 

regards, 

Lennart

**Comment by ignoramous** [comment]
Hi Lennart,

1. Can you check the *Network Log* section to see if connections were blocked (these appear with a red-colour hint) due to inadvertently turning on a Firewall setting (like blocking some app, or blocking connections that bypass DNS, etc)?
2. What Rethink version are you on?
3. Make sure Orbot is *Excluded* from the Rethink Firewall (this should happen automatically with the *Setup Orbot* flow, but there might be a bug)?
4. If you're using the *Setup Orbot* flow (from Rethink's *Settings* page), then do you choose *TCP-only Proxy* or *TCP+HTTP Proxy*?

**Comment by evilprime** [comment]
Hi thanks for your time. 

The network log section doesn't show any blocked connections. 

My rethink version is on : v053i

I've removed everything and reinstalled Rethinkdns, i've then went to the settings and installed Orbot, and follwed the process. In orbot i've started the service and went back to Rethinkdns and started the service also. I've noticed the Orbot isn't automatically added to the 'exlude' list, so i've done that myself. I then looked at the settings again in RethinkDNS for Orbot, and it was default on 'none' 

When i try Opera now, it gives me results, pages are working. Rethinkdns is saying 'protected with private dns' and the start/stop dropdown is showing its on the 'firewall' mode.

When i stop and start Rethinkdns, from the homescreen options, it crashes, every time. I restart Rethinkdns and press start and it starts again. 

i go to settings, orbot and select 'http proxy' i get 'failure starting orbot proxy etc. ' i then try it again and now i get the message 'https proxy is set' 

i now try opera again to see if webpages work, it loads, but in the end it times out. Trying different sites, no go. I look in the RethinkDNS app, and the status is telling me 'protected with tor & private dns" i go to firewall and logs, filter on block, it shows me nothing, in the 'allowed view i see the Opera app connections.  In the 'excluded' apps, i still see the Orbot app. 

I got to RethinkDNS settings > orbot and put the settings on 'none' and then internet works again. 

restarted my phone, no inet. I do get the message 'failed to start orbot as a proxy' and get the message that 'orbot stopped' trying inet and it works. Go to the RethinkDNS app and it states 'protected with private dns' so no orbot <> rethinkdns connection. 

Went to settings > orbot and selected 'http proxy' inet stops working. 

regards, 

Lennart

**Comment by ignoramous** [comment]
Do you have the *Block connections when DNS is bypassed* setting turned on, by any chance?

**Comment by evilprime** [comment]
Hi thanks for your reply. 

I have nothing on blocked in the firewall settings, though i can't find the exact rule that you are mentioning in the firewall settings?

When i look at the network logs, and filter on Orbot, i see traffic which is all 'excluded' 

I have also added the local 'android dns' settings/app(i think), the one that is being found when searching for 'dns' under the 'all apps'  to the exclusion list, you never know, though i dont see any traffic being 'blocked' 

I have checked the DNS card on the homescreen, it states that is using orbot and priavte dns, also checked the text in the bottom sheet in the rethink settings, it also states that Orbot is "forwarding TCP to Orbot, HTTP proxy also set" 

One detail i have noticed now is that when enabling everything, so Rethinkdns firewall, with Orbot and the rethink orbot settings on anything but 'none, my 'private dns' in android states that it can't connect to the nextdns host. 

it seems that in the combination Rethink+Orbot+private dns, enabling Orbot kills the connection/dns. When only using ORbot with PrivateDNS it also works. 

When just using RethinkDNS with private dns, it seems the connection to nextdns doesn't work either. Though it states on the homescreen ' protected with private dns'

**Comment by ignoramous** [comment]
What I can confirm is, Rethink `v053j` does work when Rethink is forwarding connections to Orbot with Private DNS enabled, in every device I've tried so far.

If it isn't, it most likely to do with your setup.

---

> though i can't find the exact rule that you are mentioning in the firewall settings?

It is in the *Firewall* -> (universal tab) *Block connections when DNS is bypassed*

> I have checked the DNS card on the homescreen, it states that is using orbot and priavte dns

It is likely that Orbot DNS isn't resolving Private DNS endpoint. To confirm this (leave Rethink connected to Orbot),

1. Disable Private DNS.
2. Tap on DNS and connect to *System DNS* or any other DNS upstream that you know works.
3. Enable Private DNS.
4. Check if Internet connections work.

> ...it can't connect to the nextdns host.

Use a better DNS service ;)

**Comment by ignoramous** [comment]
> Hi, looking at the Firewall universal tab, i really dont see that exact rule you mention, not even something that comes close or anything that mentions 'dns' and bypass.

You'll only see this rule if you disable Private DNS.

**Comment by ignoramous** [comment]
@evilprime we released a minor workaround in `v053k` for this issue (basically, rethink now avoids setting Orbot DNS if Private DNS is enabled). Can you see if this works (also, don't forget to check the video I linked above, just in case).

**Comment by evilprime** [comment]
Cool, will check it out tonight.

**Comment by evilprime** [comment]
ok took the time to check it out right away. I've reinstalled everything, first thing i noticed is the extra option in the stop/start drop down menu, it now has a dns+firewall mode, my version though says its on v053j, after i tried to update. 

When i leave it on dns+firewall with no private dns, i've checked the settings you discribed, its off. 

I switched on private dns, enabled tor, and stopped and started the service, it tells me its connected with TOR and Private dns. When i now try inet it works, hoereeey. One other thing i've noticed, Rethink doesn't crash anymore when i try to stop/start the service. 

Other then the version build number, all looks good!

regards, 

Lennart
